examples/amm/src/contract.rs

```diff
@@ -10,28 +10,41 @@ use amm::{AmmError, ApplicationCall, Message, Operation};
 use async_trait::async_trait;
 use fungible::{Account, FungibleTokenAbi};
 use linera_sdk::{
-    base::{AccountOwner, Amount, ApplicationId, Owner, WithContractAbi},
+    base::{AccountOwner, Amount, ApplicationId, WithContractAbi},
     contract::system_api,
     ensure, ApplicationCallOutcome, Contract, ContractRuntime, ExecutionOutcome, OutgoingMessage,
     Resources, ViewStateStorage,
 };
 use num_bigint::BigUint;
 use num_traits::{cast::FromPrimitive, ToPrimitive};
 
-linera_sdk::contract!(Amm);
+pub struct AmmContract {
+    state: Amm,
```

Comment:
Should this be `AmmState`?

Reply:
Yeah, maybe it makes sense to rename all the state types to have the suffix 🤔